core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -144,7 +146,18 @@ public static <T extends StructLike> PositionDeleteIndex toPositionIndex(
             deletes ->
                 CloseableIterable.transform(
                     locationFilter.filter(deletes), row -> (Long) POSITION_ACCESSOR.get(row)));
-    return toPositionIndex(CloseableIterable.concat(positions));
+    return toPositionIndex(positions);
+  }
+
+  public static PositionDeleteIndex toPositionIndex(List<CloseableIterable<Long>> positions) {
```

Comment:
Can you have this check whether there is more than one iterable? There's no need to use ParallelIterable if there's only one.

Reply:
Thanks @rdblue. Yes, this happens when there are more than one "delete positional file" that qualifies for the data file. E.g Assume a trickle feed job ingests data into the partition. Due to late arriving data, another job updates the dataset for certain dataset in the partition & creates "positional files (POS)".  For update jobs with different criteria, same data file may get qualified and creates additional POS files.  Essentially during scanning, one data file may have to scan multiple POS files (e.g 4 pos files) and causes slowness. ParallelIterable helps in this case.

Reply:
Yes, I understand the purpose of the PR. Here, there should be a check so that ParallelIterable is only used when there is more than one delete file to read.

Reply:
Addressed it in the recent PR. Also, made it as a configurable option via sys properties.